packages/mui-base/src/Dialog/Root/DialogRoot.types.ts

```diff
@@ -0,0 +1,112 @@
+export type DialogType = 'dialog' | 'alertdialog';
+
+export type SoftCloseOptions = boolean | 'clickOutside' | 'escapeKey';
+
+export interface DialogRootProps extends UseDialogRootParameters {
+  children?: React.ReactNode;
+}
+
+export interface UseDialogRootParameters {
+  /**
```

Comment:
I think we may be missing a `lockScroll` prop. If it's modal the body scroll should be locked. [Here is a technique that works on iOS](https://github.com/floating-ui/floating-ui/blob/3656e71dbd59078debc393696a50b4b5aee57f65/packages/react/src/components/FloatingOverlay.tsx#L31) which we could use.

Reply:
I added the useScrollLock utility, but it seems it works on our docs only when the styles are applied on the html element, not the body. Do you know why it does it behave this way?

Reply:
It just `overflow: hidden` to the `<body>` for most devices, sans iOS, which is the usual technique for locking body scroll. Not sure why that wouldn't work.